contracts/standard/proxy/ArbitratorVersioningProxy.sol

```diff
@@ -37,29 +42,34 @@ contract ArbitratorVersioningProxy is Arbitrator, VersioningProxy {
      *  @param _extraData Can be used to give additional info on the dispute to be created.
      *  @return The ID of the created dispute in the context of this contract.
      */
-    function createDispute(uint256 _choices, bytes _extraData) public payable returns(uint256 _disputeID) {
-        uint256 _arbitratorDisputeID = Arbitrator(implementation).createDispute.value(msg.value)(_choices, _extraData);
-        return disputes.push(
-            Dispute({
+    function createDispute(uint256 _choices, bytes _extraData) public payable returns(uint256 _localDisputeID) {
+        uint256 externalDisputeID = Arbitrator(implementation).createDispute.value(msg.value)(_choices, _extraData);
+        _localDisputeID = disputes.push(
+            DisputeStruct({
+                arbitrated: Arbitrable(msg.sender),
+                externalDisputeID: externalDisputeID,
                 arbitrator: Arbitrator(implementation),
-                disputeID: _arbitratorDisputeID,
                 choices: _choices
             })
         );
+
+        externalDisputeIDToLocalDisputeID[externalDisputeID] = _localDisputeID;
     }
 
     /** @notice Appeals a ruling to the current `implementation` contract.
-     *  @param _disputeID The ID of the dispute to be appealed.
+     *  @param _localDisputeID The ID of the dispute to be appealed.
      *  @param _extraData Can be used to give extra info on the appeal.
      */
-    function appeal(uint256 _disputeID, bytes _extraData) public payable {
-        if (disputes[_disputeID].arbitrator != implementation) { // Arbitrator has been upgraded, create a new dispute in the new arbitrator
-            uint256 _choices = disputes[_disputeID].choices;
+    function appeal(uint256 _localDisputeID, bytes _extraData) public payable {
+        if (disputes[_localDisputeID].arbitrator != implementation) { // Arbitrator has been upgraded, create a new dispute in the new arbitrator
+            Arbitrable _arbitrated = disputes[_localDisputeID].arbitrated;
+            uint256 _choices = disputes[_localDisputeID].choices;
             uint256 _arbitratorDisputeID = Arbitrator(implementation).createDispute.value(msg.value)(_choices, _extraData);
-            disputes[_disputeID] = Dispute({ arbitrator: Arbitrator(implementation), disputeID: _arbitratorDisputeID, choices: _choices });
+            disputes[_localDisputeID] = DisputeStruct({ arbitrated: _arbitrated, arbitrator: Arbitrator(implementation), externalDisputeID: _arbitratorDisputeID, choices: _choices });
```

Comment:
It would be cheaper to only overwrite `arbitrator` and `externalDisputeID`, instead of creating a new struct.